ConTabs/OutputBuilder.cs

```diff
@@ -140,6 +156,44 @@ private void DataLine(List<CellParts> parts, int line)
                 }
             }
 
+            private int GetWidthOfPaddingAndBorders()
+            {
+                return table._colsShown.Count * table.Padding.GetHorizontalPadding() + table._colsShown.Count + 1;
+            }
+
+            private int GetTableWidthAfterApplyingStretchStyles()
+            {
+                int colWidths = 0;
+                foreach (Column column in table._colsShown)
+                {
+                    int colWidth = table.TableStretchStyles.CalculateOptimalWidth(column);
+                    column.LongStringBehaviour.DisplayWidth = colWidth;
+                    colWidths += colWidth;
+                }
+                int realWidth = colWidths + GetWidthOfPaddingAndBorders();
+                int adaptedWidth = table.TableStretchStyles.CalculateAdditionalWidth(table._colsShown, realWidth, table.CanvasWidth);
+                return adaptedWidth + GetWidthOfPaddingAndBorders();
+            }
+
+            private int GetOptimalTableWidth(List<Column> hiddenCols)
```

Comment:
Passing `hiddenCols` as an argument here feels a bit off. This method is an instance member of the `Table` class, so it can access the list of columns directly, without needing to pass around additional lists.